app/Actions/CodeGenerator.php

```diff
@@ -2,12 +2,25 @@
 
 namespace App\Actions;
 
+use App\Models\ShortUrl;
 use Illuminate\Support\Str;
 
 class CodeGenerator
 {
+    //definindo um construtor para chamar a classo passando length diferente nos testes por exemplo
+    public function __construct(
+        private int $length = 0
+    )
+    {
+        $this->length = $length ?? config('app.short_url_code_length');
+    }
+
     public function run(): string
     {
-        return Str::random(5);
+        $code = Str::random($this->length);
+
+        return ShortUrl::where('code', $code)->count()
```

Comment:
```suggestion
        return ShortUrl::where('code', $code)->exists()
```

Mudando para `exists()` para melhorar a leitura do código.